ctl/src/command.rs

```diff
@@ -427,26 +424,23 @@ pub fn status(mut channel: Channel<CommandRequest,CommandResponse>, json: bool)
 
   match channel.read_message() {
     None          => {
-      eprintln!("the proxy didn't answer");
-      exit(1);
+      bail!("the proxy didn't answer");
     },
     Some(message) => {
       if id != message.id {
-        eprintln!("received message with invalid id: {:?}", message);
-        exit(1);
+        bail!("received message with invalid id: {:?}", message);
       }
       match message.status {
         CommandStatus::Processing => {
-          eprintln!("should have obtained an answer immediately");
-          exit(1);
+          bail!("should have obtained an answer immediately");
         },
         CommandStatus::Error => {
           if json {
-            print_json_response(&message.message);
+            print_json_response(&message.message)?;
+            Ok(())
```

Comment:
Same here, you should return an error

Reply:
I tried the same as above to correct it.